_stbt/frameobject.py

```diff
@@ -413,3 +413,56 @@ def is_visible(self):
         raise NotImplementedError(
             "Objects deriving from FrameObject must define an is_visible "
             "property")
+
+    def refresh(self, frame=None, **kwargs):
+        r"""Returns a new FrameObject typically of the same type as `self`, but
+        with un updated frame.  `self` is not modified.
+
+        If you want to override this behaviour implement `_refresh` on your
+        derived class instead.
+
+        .. testsetup::
+
+            >>> import stbt
+            >>> from tests.test_frame_object import _load_frame
+            >>> class Dialog(stbt.FrameObject):
+            ...     @property
+            ...     def is_visible(self):
+            ...         return bool(self._info)
+            ...
+            ...     @property
+            ...     def title(self):
+            ...         return stbt.ocr(region=stbt.Region(396, 249, 500, 50),
+            ...                         frame=self._frame)
+            ...
+            ...     @property
+            ...     def message(self):
+            ...         right_of_info = stbt.Region(
+            ...             x=self._info.region.right, y=self._info.region.y,
+            ...             width=390, height=self._info.region.height)
+            ...         return stbt.ocr(region=right_of_info, frame=self._frame) \
+            ...                .replace('\n', ' ')
+            ...
+            ...     @property
+            ...     def _info(self):
+            ...         return stbt.match('tests/info.png', frame=self._frame)
+
+        >>> page = Dialog(frame=_load_frame('with-dialog'))
+        >>> print page.message
+        This set-top box is great
+        >>> page = page.refresh(_load_frame('with-dialog2'))
+        >>> print page.message
+        This set-top box is fabulous
+
+        Added in v30"""
+        if frame is None:
+            import stbt
+            frame = stbt.get_frame()
+        return self._refresh(frame=frame, **kwargs)
+
+    @classmethod
+    def _refresh(cls, frame):
```

Comment:
Answers: 

2. Yes, with `type(self)` or `self.__class__`.
3. Yes, a normal method in the sub-class can override a class method in the super-class.

Reply:
> Why is _refresh a classmethod? Answer: Because it takes a cls which you can use to create an instance of the same class.

Indeed.  I suppose the idea was to provide additional flexibility by minimizing the requirements on the caller.  But in reality the caller doesn't have the option of calling this with just a class rather than an instance because an explicit goal of this it to allow the implementer to consider `self` when refreshing.  So I think this can be changed to `type(self)(frame=frame)` and it will make more sense.